ipi/engine/motion/replay.py

```diff
@@ -58,7 +61,34 @@ def __init__(self, fixcom=False, fixatoms=None, intraj=None):
             raise ValueError(
                 "Replay can only read from PDB or XYZ files -- or a single frame from a CHK file"
             )
-        self.rfile = open(self.intraj.value, "r")
+        # Posibility to read beads from separate XYZ files by a wildcard
+        if any(char in self.intraj.value for char in "*?[]"):
+            infilelist = []
+            for file in sorted(os.listdir('.')):
+                if fnmatch(file, self.intraj.value):
+                    infilelist.append(file)
+            # determine bead numbers in input files
+            bead_map_list = []
+            for file in infilelist:
+                fdin = open(file, 'r')
+                rr = read_file_raw(self.intraj.mode, fdin)
+                metainfo = rr['comment'].split()
+                for i, word in enumerate(metainfo):
+                    if word == 'Bead:':
+                        bead_map_list.append(int(metainfo[i + 1]))
+                fdin.close()
+            # check that beads are continuous (no files missing)
+            if sorted(bead_map_list) != list(range(len(bead_map_list))):
+                raise ValueError(
+                    "Provided trajectory files have non-continuous "
+                    "range of bead indices."
+                )
+            # sort the list of files according to their bead indices
+            infilelist_sorted, _ = zip(*sorted(zip(infilelist, bead_map_list),
```

Comment:
why do you need to sort?

Reply:
Because then I do a loop and initialize beads from the files, and I'm not sure whether os.listdir will always return files in correct order, even when sorted by filename (it should in simple cases, but people can always break it somehow).

Generally, I think that keeping the order is a nice feature, because it gives uniformity of outputs. I assume that the beads for a replay should all come from the same simulation. It's either a full set, or a sparse subset of the original beads. The idea is to make the setup stable for 99% of most used cases, and the rest 1% of people can adjust their files.

Alternatively, I can order files only in case when the provided beads are continuous (which is the only option now), and keep arbitrary order otherwise.

Reply:
I decided to sort always. I don't see where it can harm, and I find it anyways better than relying on sorting of the filenames.